### ISSUE TITLE
BUG - '/' as part of key not possible 

### ISSUE THREAD

**Issue by DanielJakobs** [SAP/open-ux-odata#276]
Dear colleagues,
seems that the mockserver is not able to find an entry when the key contains a '/'.
Our test data:
    {
      "Name": "/ABC/TEST",
      "GlobalAccountID": "",
      "ProductID": "",
      "Vendor": "",
      "Namespace": "/ABC/",
      "filterfield": false
    },

our metadata.xml:
```
            <EntityType Name="xA4CLPxCICD_ADDON_PRODUCT_CEType">
                <Key>
                    <PropertyRef Name="Name"/>
                </Key>
                <Property Name="Name" Type="Edm.String" Nullable="false" MaxLength="30"/>
                <Property Name="GlobalAccountID" Type="Edm.String" Nullable="false" MaxLength="200"/>
                <Property Name="ProductID" Type="Edm.String" Nullable="false" MaxLength="20"/>
                <Property Name="Vendor" Type="Edm.String" Nullable="false" MaxLength="20"/>
                <Property Name="Namespace" Type="Edm.String" Nullable="false" MaxLength="10"/>
                <Property Name="filterfield" Type="Edm.Boolean" Nullable="false"/>
```
...

my testing with javascript for the odata v4 model:
```
      const oModel = this.getView().getModel('addon_create_vers_srv');

      const oRequestSingleEntry = oModel.bindContext("/xA4CLPxCICD_ADDON_PRODUCT_CE('%2FABC%2FTEST')");

      oRequestSingleEntry.requestObject().then((oObject) => {
        const { Name: sName, Namespace: sNamespace } = oObject;
        console.log(`${sName} ${sNamespace}`);
      });
```

it fails with:
Failed to read path /xA4CLPxCICD_ADDON_PRODUCT_CE('%2FABC%2FTEST') - Error: Cannot read properties of undefined (reading 'entityType')

We use '/' in our key because the product name contain a namespace.

When converting the whole mockdata .json files to use %2F instead of '/', then it works perfectly, but on the UI I see the encoded values like %2FITAPC%2FBONUS_MANAGEMENT, pretty anoying ;-).

The rap framework is not having this issue and seems to encode/decode '/' correctly.

Greetings,
Daniel
You can use my sap email adress and I could send you a link to our git with that example

**Comment by nlunets** [comment]
Slash keys are indeed not supported :) 
Seem like an easy fix, will have a look shortly

**Comment by nlunets** [comment]
More accurately if you request them with `/xA4CLPxCICD_ADDON_PRODUCT_CE('%2FABC%2FTEST')` then it is easy to support, if you also need `/xA4CLPxCICD_ADDON_PRODUCT_CE('/ABC/TEST')` to work then that's a bit more work since we'd need to rewrite our query parser :)

**Comment by DanielJakobs** [comment]
Hi,
/xA4CLPxCICD_ADDON_PRODUCT_CE('%2FABC%2FTEST') is fair enough. That's how we get used to it, when using the old world odatamodel.read(...)

thx

**Comment by nlunets** [comment]
Merged as part of #277, will be available in version 2.1.34
